### ISSUE TITLE
support one-sample test in `ggbarstats`

### ISSUE THREAD

**Issue by IndrajeetPatil** [IndrajeetPatil/ggstatsplot#532]
if `y` is missing, create an internal column with a unique value?

not sure how happy the `statsExpressions` would be with this behavior

**Comment by AntoineSoetewey** [comment]
Hello,

I see that one-proportion and chi-square goodness of fit tests are possible with ggpiestats().

I am not a big fan of pie charts, I'd rather use bar charts (and I believe I am not alone). By any chance, is there any plan to include the exact same tests but with bar charts instead of pie charts?

Thanks again for the package!

**Comment by IndrajeetPatil** [comment]
> is there any plan to include the exact same tests but with bar charts instead of pie charts

That is exactly what the current issue is about! :)

The problem is that a bar chart is an inherently bivariate visualization, while a pie chart a univariate one. So one-way table tests play nicely with the latter but not the former. 

When I work on this, my approach is probably going to be to create a temporary internal variable so that the bar chart is happy and one-way table tests can be carried out.

**Comment by AntoineSoetewey** [comment]
Thanks for your reply.

Just one quick question :

« The problem is that a bar chart is an inherently bivariate visualization »
—> I thought barplot(table(data$variable)) was considered as univariate, isn’t the case?

**Comment by IndrajeetPatil** [comment]
Good point! You're right that a bar chart *in general* (e.g., `barplot(table(data$variable))`) is univariate — it shows the distribution of a single variable.

What was meant here is that `ggbarstats()` specifically creates **stacked** bar charts, which map one variable to the x-axis and another to the fill/stacking. That particular visualization is inherently bivariate, which is why the function currently requires both `x` and `y`.

But there's no reason a non-stacked bar chart couldn't display one-sample test results the same way `ggpiestats()` does. That's what this issue is tracking.
